java/flight/flight-sql-jdbc-core/src/main/java/org/apache/arrow/driver/jdbc/accessor/impl/calendar/ArrowFlightJdbcTimeStampVectorAccessor.java

```diff
@@ -96,7 +96,7 @@ private LocalDateTime getLocalDateTime(Calendar calendar) {
       TimeZone timeZone = calendar.getTimeZone();
       long millis = this.timeUnit.toMillis(value);
       localDateTime = localDateTime
-          .minus(timeZone.getOffset(millis) - this.timeZone.getOffset(millis), ChronoUnit.MILLIS);
+          .plus(timeZone.getOffset(millis) - this.timeZone.getOffset(millis), ChronoUnit.MILLIS);
```

Comment:
The original buggy code I want to solve is here. It tries to convert local date time from vector timezone to user-supplied timezone. We should add the timezone offset but not subtract it. A number of test cases need to be fixed as their expected values are computed with the same issue.